src/ERC3009.sol

```diff
@@ -48,18 +115,90 @@ abstract contract ERC3009 is IERC3009, StatefulERC712 {
         bytes32 r_,
         bytes32 s_
     ) external {
-        _transferWithAuthorization(
-            TRANSFER_WITH_AUTHORIZATION_TYPEHASH,
+        _transferWithAuthorizationBefore(from_, validAfter_, validBefore_, nonce_);
+
+        _revertIfInvalidSignature(
+            from_,
+            _getDigest(
+                keccak256(
+                    abi.encode(
+                        TRANSFER_WITH_AUTHORIZATION_TYPEHASH,
+                        from_,
+                        to_,
+                        value_,
+                        validAfter_,
+                        validBefore_,
+                        nonce_
+                    )
+                )
+            ),
+            abi.encodePacked(r_, s_, v_)
+        );
+
+        _transferWithAuthorizationAfter(from_, to_, value_, nonce_);
+    }
+
+    /// @inheritdoc IERC3009
+    function receiveWithAuthorization(
+        address from_,
+        address to_,
+        uint256 value_,
+        uint256 validAfter_,
+        uint256 validBefore_,
+        bytes32 nonce_,
+        bytes memory signature_
+    ) external {
+        _revertIfInvalidSignature(
             from_,
-            to_,
-            value_,
-            validAfter_,
-            validBefore_,
-            nonce_,
-            v_,
-            r_,
-            s_
+            _getDigest(
+                keccak256(
+                    abi.encode(
+                        RECEIVE_WITH_AUTHORIZATION_TYPEHASH,
+                        from_,
+                        to_,
+                        value_,
+                        validAfter_,
+                        validBefore_,
+                        nonce_
+                    )
+                )
+            ),
+            signature_
         );
+
+        _receiveWithAuthorization(from_, to_, value_, validAfter_, validBefore_, nonce_);
+    }
+
+    /// @inheritdoc IERC3009
+    function receiveWithAuthorization(
+        address from_,
+        address to_,
+        uint256 value_,
+        uint256 validAfter_,
+        uint256 validBefore_,
+        bytes32 nonce_,
+        bytes32 r_,
+        bytes32 vs_
+    ) external {
+        _revertIfInvalidSignature(
+            from_,
+            _getDigest(
+                keccak256(
+                    abi.encode(
+                        RECEIVE_WITH_AUTHORIZATION_TYPEHASH,
+                        from_,
+                        to_,
+                        value_,
+                        validAfter_,
+                        validBefore_,
+                        nonce_
+                    )
+                )
+            ),
+            abi.encodePacked(r_, vs_)
```

Comment:
Same here.

Reply:
Fixed in this commit: https://github.com/MZero-Labs/common/pull/8/commits/b63a1861b99ff78cc8356262b8076badf40154a4